packages/tocco-ui/src/SignalList/SignalListItem.js

```diff
@@ -0,0 +1,53 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import Icon from '../Icon'
+import StyledSignalListItem from './StyledSignalListItem'
+import {
+  conditionPropTypes,
+  stylingPosition,
+  stylingCondition
+} from '../utilStyles'
+
+const getIcon = props => {
+  switch (props.condition) {
+    case stylingCondition.DANGER:
+      return {icon: 'fa-times'}
+    case stylingCondition.SUCCESS:
+      return {icon: 'fa-check'}
+    case stylingCondition.WARNING:
+      return {icon: 'fa-exclamation-triangle'}
+    case stylingCondition.PRIMARY:
+    case stylingCondition.BASE:
+      return {unicode: '\u2022'} // unicode bullet point
```

Comment:
what if a new styling condition item is added at some point? maybe it would make sense to return the unicode bullet point as default?

Reply:
I share your point of view that it would make sense to consider it here if a new condition is added. However, in my opinion, it's more important to make our code as stable as possible - chances are that we won't notice that a new styling condition has been added until this component looks broken in the production environment, which is certainly not what we want to achieve.

Instead of replacing the `PRIMARY` and `BASE` cases with a `default` block, we could keep them there and add a `default` block where we return the unicode bullet point but also log a warning in the console. This way, the risk of a broken production environment would be reduced, but we would still notice that a new styling condition has been added which should be considered in this component.

Reply:
I've thought about this a little bit more and I've come up with another idea.

What if we would define the icons in a map instead of returning them in a switch/case. This map could be compared with the prop types when the file is loaded. This way we could log the warning when the file is loaded instead of when the component is rendered (which could be discovered rather late, maybe already in production).

```
const ICONS = {
  [stylingCondition.DANGER]: {icon: 'fa-times'},
  [stylingCondition.SUCCESS]: {icon: 'fa-check'},
  ...
}

requireAllKeys(ICONS, conditionPropTypes)

// can be somewhere in the utils module
function requireAllKeys(map, propType) {
  // compare map with prop type items and log if not the same
}

const getIcon = props => {
  const icon = ICONS[props.stylingCondition]
  return icon || {unicode: '\u2022'}
}